File: 66.plus-one.py
# @leet start
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1
        for i in range(len(digits) - 1, -1, -1):
            current = digits[i] + carry
            digits[i] = current % 10
            if current // 10 > 0:
                carry = 1
            else:
                carry = 0
        if carry == 1:
            digits.insert(0, carry)
        return digits


# @leet end

File: 66.plus-one.cpp
// @leet start
#include <vector>
class Solution {
public:
  vector<int> plusOne(vector<int> &digits) {
    int carry = 1;
    for (int i = digits.size() - 1; i >= 0; i--) {
      int current = carry + digits[i];
      digits[i] = current % 10;

      if (current >= 10) {
        carry = 1;
      } else {
        carry = 0;
      }
    }
    if (carry) {
      digits.insert(digits.begin(), carry);
    }
    return digits;
  }
};
// @leet end
